### PR TITLE
Two minor fixes to unsafe code

### PR DESCRIPTION
**Pull request by brson**
### What problem does this PR solve?

Two fixes found during casual code review. The first replaces an unnecessary version of `from_utf8_unchecked` with the checked version. The second adds error handling to the foreign `setmntent` call.

### Release note

### CODE REVIEW

File: components/tikv_util/src/config.rs [hunk 1]
```diff
@@ -398,7 +398,7 @@ impl<'de> Deserialize<'de> for ReadableDuration {
                         ));
                     }
                     // do we need to check 12h360m?
-                    let number_str = unsafe { str::from_utf8_unchecked(first) };
+                    let number_str = str::from_utf8(first).expect("utf8");
```
Comment (BusyJay) [inline components/tikv_util/src/config.rs:401]: But apparently first has to be encoded in utf-8, why add an unnecessary check?

Comment (brson) [reply components/tikv_util/src/config.rs:401]: Performance doesn't matter here. We shouldn't be writing unsafe code unless necessary.

Comment (BusyJay) [reply components/tikv_util/src/config.rs:401]: What's the harm when we know for sure it's safe to write an unsafe?

Comment (brson) [reply components/tikv_util/src/config.rs:401]: People too often do not know for sure that it is safe to write unsafe, even if they think they know. By not writing 'unsafe' at all it is obviously not unsafe. Every appearance of unsafe requires that some invariant must hold, not just now, but in the future as other developers maintain that unsafe code.

Comment (brson) [reply components/tikv_util/src/config.rs:401]: FWIW, this is just one case of unnecessary unsafe, and whether or not it is removed doesn't matter much. I think the TiKV codebase has much too much unsafe code though.

Comment (brson) [reply components/tikv_util/src/config.rs]: For now I've removed the commit that removes the use of `from_utf8_unchecked`. I may return to this issue more generally in the future.

**Pull request merged**